internal/service/resources/api/server.go

```diff
@@ -256,32 +212,158 @@ func (r *ResourceServer) DeleteSubscription(ctx context.Context, request api.Del
 
 // GetResourcePools receives the API request to this endpoint, executes the request, and responds appropriately
 func (r *ResourceServer) GetResourcePools(ctx context.Context, request api.GetResourcePoolsRequestObject) (api.GetResourcePoolsResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	records, err := r.Repo.GetResourcePools(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to retrieve resource pools: %w", err)
+	}
+
+	objects := make([]api.ResourcePool, len(records))
+	for i, record := range records {
+		objects[i] = models.ResourcePoolToModel(&record)
+	}
+
+	return api.GetResourcePools200JSONResponse(objects), nil
 }
 
 // GetResourcePool receives the API request to this endpoint, executes the request, and responds appropriately
 func (r *ResourceServer) GetResourcePool(ctx context.Context, request api.GetResourcePoolRequestObject) (api.GetResourcePoolResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	records, err := r.Repo.GetResourcePool(ctx, request.ResourcePoolId)
+	if err != nil {
+		return api.GetResourcePool500ApplicationProblemPlusJSONResponse{
+			AdditionalAttributes: &map[string]string{
+				"resourcePoolId": request.ResourcePoolId.String(),
+			},
+			Detail: err.Error(),
+			Status: http.StatusInternalServerError,
+		}, nil
+	}
+
+	if len(records) == 0 {
+		return api.GetResourcePool404ApplicationProblemPlusJSONResponse{
+			AdditionalAttributes: &map[string]string{
+				"resourcePoolId": request.ResourcePoolId.String(),
+			},
+			Detail: "requested resourcePool not found",
+			Status: http.StatusNotFound,
+		}, nil
+	}
+
+	object := models.ResourcePoolToModel(&records[0])
+	return api.GetResourcePool200JSONResponse(object), nil
 }
 
 // GetResources receives the API request to this endpoint, executes the request, and responds appropriately
 func (r *ResourceServer) GetResources(ctx context.Context, request api.GetResourcesRequestObject) (api.GetResourcesResponseObject, error) {
-	// TODO implement me
-	return nil, fmt.Errorf("not implemented")
+	// First, find the pool
+	pools, err := r.Repo.GetResourcePool(ctx, request.ResourcePoolId)
```

Comment:
what about using select exists for this? We just need a quick way to check that it is there
https://github.com/jackc/pgx/discussions/1427

Reply:
Not sure I follow the intent of your comment.  Are you proposing that we add a different repo API that essentially is a `exists, err := ResourceExists(ctx, request.ResourcePoolId)`

Reply:
If so, I'd rather tackle that at another time.

Reply:
No, that's not what I meant. Here and in the function below, you are checking if the `pool` exists. You do that by using `Search` which fetches all the columns. Nothing is done with the data of the pool.

So, what I try to say is that there is a more efficient way to do this. We could create an utility function which checks with `SELECT EXISTS` if an object exists

Reply:
fixed